File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int] ,  nums2:List[int]) -> List[int]:
        nums1, nums2 = sorted(nums1) , sorted(nums2)
        res = []
        i, j= 0,0
        while i< len(nums1) and j <len(nums2):
            if nums1[i] < nums2[j]:
                i+=1
            elif nums1[i] > nums2[j]:
                j+=1
            else:
                res.append(nums1[i])
                i+=1
                j+=1
        return res

    def intersect2(self, nums1: List[int] , nums2: List[int]) -> List[int]:
        c = Counter(nums1)
        res =[]
        for n in nums2:
            if c[n] >0:
                res.append(n)
                c[n] -=1
        return res

    def intersect1(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        for n in nums1:
            if n in nums2:
                res.append(n)
                nums2.remove(n)
        return res

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1,  nums2){
    nums1 = nums1.sort((a,b) => a -b)
    nums2 = nums2.sort((a,b) => a- b)
    let i= 0
    let j = 0
    let res = []
    while (i<nums1.length && j < nums2.length){
        if (nums1[i] < nums2[j]) i++
        else if (nums1[i] > nums2[j]) j++
        else {
            res.push(nums1[i])
            i++
            j++
        }
    }
    return res
}
var intersect1 = function(nums1, nums2) {
    let res = []
    for (let n of nums1){
        if (n in nums2){
            res.push(n)
            nums2= nums2.filter(function(item){
                                return item !== n
                                })
        }
    }
    return res
};